### ISSUE TITLE
Unsafe code: Added redundant Cast

### ISSUE THREAD

**Issue by bprg** [icsharpcode/ILSpy#1021]
Code
```
byte* pByte = ...
pByte += 4;
```

Decompile by ILSpy
```
byte* pByte = ...
pByte += (byte*)4;
```

**Comment by dgrunwald** [comment]
Can you provide a more complete code example?

I tried:
```
		static unsafe byte* Test(byte* pByte)
		{
			pByte += 4;
			return pByte;
		}
```
and that gets decompiled correctly.

**Comment by bprg** [comment]
Please try the following code, Thanks for your support.
```
		public static unsafe void Test(ref byte* ptr)
		{
			ptr += 4;
		}
```

**Comment by dgrunwald** [comment]
Ah of course, the simple local variable I tried is a special case and does not use the `CompoundAssignmentInstruction` in the ILAst.
The `CompoundAssignmentInstruction`->C# translation was completely missing the case for pointer types. It wasn't just incorrect casts, with struct pointers we would emit:
`structPtr += (MyStruct*)(offset * sizeof(MyStruct));`
even though the multiplication by the element size should be removed during decompilation. This is now fixed.
